plutus-scb/src/Cardano/Metadata/Client.hs

```diff
@@ -1,40 +1,59 @@
-{-# LANGUAGE GADTs            #-}
-{-# LANGUAGE LambdaCase       #-}
-{-# LANGUAGE RankNTypes       #-}
-{-# LANGUAGE TypeApplications #-}
-{-# LANGUAGE TypeOperators    #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE KindSignatures      #-}
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# LANGUAGE TypeOperators       #-}
 
-module Cardano.Metadata.Client where
+module Cardano.Metadata.Client
+    ( handleMetadataClient
+    ) where
 
 import           Cardano.Metadata.API      (API)
-import           Cardano.Metadata.Types    (MetadataEffect (GetProperties, GetProperty), Property, PropertyKey, Subject)
+import           Cardano.Metadata.Types    (JSONEncoding (ExternalEncoding),
+                                            MetadataEffect (GetProperties, GetProperty),
+                                            MetadataError (MetadataClientError))
 import           Control.Monad.Freer       (Eff, LastMember, Member, type (~>), interpret, sendM)
 import           Control.Monad.Freer.Error (Error, throwError)
 import           Control.Monad.IO.Class    (MonadIO, liftIO)
+import           Data.Coerce               (coerce)
 import           Data.Proxy                (Proxy (Proxy))
-import           Servant.Client            (ClientEnv, ClientError, ClientM, client, runClientM)
+import           Network.HTTP.Types        (status404)
+import           Servant.Client            (ClientEnv, ClientError (FailureResponse), ClientM, client,
+                                            responseStatusCode, runClientM)
 import           Servant.Extra             (left, right)
 
-getProperties :: Subject -> ClientM [Property]
-getProperty :: Subject -> PropertyKey -> ClientM Property
-(getProperties, getProperty) = (_getProperties, _getProperty)
-  where
-    _getProperties = left . api
-    _getProperty = right . api
-    api = client (Proxy @API)
-
 handleMetadataClient ::
        forall m effs.
-       (LastMember m effs, MonadIO m, Member (Error ClientError) effs)
+       ( LastMember m effs
+       , MonadIO m
+       , Member (Error MetadataError) effs
+       )
     => ClientEnv
     -> Eff (MetadataEffect ': effs) ~> Eff effs
 handleMetadataClient clientEnv =
-    let runClient :: forall a. ClientM a -> Eff effs a
-        runClient a =
-            (sendM $ liftIO $ runClientM a clientEnv) >>= either throwError pure
+    let (getProperties, getProperty) = (_getProperties, _getProperty)
+          where
+            _getProperties = left . api
+            _getProperty = right . api
+            -- This can be flipped between AesonEncoding and
+            -- ExternalEncoding depending on which metadata server
+            -- we're using.
+            api = client (Proxy @(API 'ExternalEncoding))
```

Comment:
why not
```haskell
let api = client (Proxy @(API 'ExternalEncoding))
    getProperties = left . api
    getProperty = right . api
```
the use of `let` and `where` is weird

Reply:
Agreed & fixed.